### PR TITLE
New Giant Mole Plugin & Various fixes and improvements  

### PR DESCRIPTION
**Pull request by See1Duck**
# Announcement: **New Giant Mole Plugin**

Im thrilled to introduce my **brand-new Giant Mole Plugin**, a comprehensive automation solution for hunting the Giant Mole! Here’s what you can expect:

- **State-Based Logic**: The GiantMoleScript handles activities such as combat, looting, banking, and world-hopping in a highly efficient, step-by-step manner.
- **Real-Time Overlay**: The GiantMoleOverlay displays the mole’s location, your current bot state, and other critical details right on your screen.
- **Configurable Settings**: Through the GiantMoleConfig panel, you can customize looting preferences, set inventory setups, and fine-tune automation behaviors.
- **Safety Checks & Fallbacks**: Enjoy built-in mechanisms for detecting occupied worlds, using prayers, and consuming potions to keep your hunts smooth and secure.

This plugin brings streamlined mole hunting to the next level! Let the pet hunting begin!

# Changelog

## Spellbook & Magic Improvements (Rs2Magic)
- **Simplified spellbook filter toggling** to reduce overhead and streamline script performance.
- **Improved magic casting logic** by making spellbook checks more efficient and accessible, reducing repetitive checks and enhancing responsiveness.
- **Added quick-cast functionality** for faster, more human-like spell usage.

## Botting & Quality of Life (QoL) Updates
- **Enhanced botting plugins** with new firemaking behavior that respects user-configured quick-firemaking settings.
- **Case-insensitive bank chest detection** for more reliable banking automation.

## Smart Automation for Runecrafting (QoL)
- **Added “Smart Workbench”** for Guardian of the Rift, enabling one-click pouch filling and automatic crafting.
- **Enabled “Smart Mine”** to mine Huge Guardian Remains with one click for easy pouch filling.
- **Introduced “Smart Runecraft”** to auto-craft runes at Altars, handling pouches and crafting in a single click.

## Antiban & Logging
- **Refined AntibanPlugin logs** to only show activity changes when dev debugging is enabled, reducing clutter for regular users.
- **Introduced mouse speed intensity controls** with a slider to allow users to adjust mouse speed for more or less human-like behavior.
- **Defaulted activity intensity to EXTREME** on startup and reset for more robust antiban behavior.

## GabulhasGlassMake Bot Enhancements
- **Added script version display** on the overlay for clearer tracking.
- **Implemented universal antiban setup** and adjusted cooldowns to simulate human timing.
- **Optimized spellbook checks** and quick-cast usage for more efficient casting.
- **Shortened the bot delay interval** for faster processing.
- **Reset antiban settings** upon script shutdown for consistent behavior.

## Runestone Interaction Tweaks (Arcceus RC)
- **Reduced proximity range** to runestones for more precise movement.
- **Increased interaction range** when mining dense runestones to improve gathering efficiency.

## Thieving (Antiban)
- **Fixed thieving bot busy-state detection** to prevent idle-mode mistakes.

## Inventory Setup Integration
- **Added inventory setup selection** in the Microbot configuration panel, allowing quick loading of saved setups.
- **Refactored loadout handling** to use dedicated InventorySetup objects for better stability and error prevention.
- **Implemented serialization** for InventorySetup, ensuring reliable saving and loading of configurations.

## High Alch, Banking & Prayer (AIO Fighter)
- **Added a High Alch script** that automatically identifies and alchs profitable items, with a toggle for easy activation.
- **Enhanced banking logic** to properly resume idle states after banking, improving bot flow.
- **Introduced profitability checks** for High Alch to ensure items are only alched when profitable.
- - **Improved prayer activation** by checking both combat status and NPC tracking, ensuring more accurate quick-prayer toggling.

## Stamina & Potion Usage (QoL)
- **Adjusted stamina threshold multiplier** for accurate stamina potion usage based on configured settings.
- **Recognized additional potion variants** (combat, super combat, magic, divine) to improve automated potion handling.
- **Implemented automated combat potion usage** for optimized stat boosts during fights.

## Other Notable Changes
- **Made boss timer and respawn timer classes public** for better extensibility with other plugins.
- **Fixed High Alch condition logic** to avoid alching when the profitability setting isn’t met.
- **Removed unnecessary logging** in PrayerScript to reduce clutter.
- **Added a Giant Mole plugin** with state-based automation, including combat, hunting, banking, and world-hopping.
- **Handled empty inventory setups** gracefully to prevent errors in the bot configuration panel.

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/microbot/util/settings/Rs2SpellBookSettings.java [hunk 1]
```diff
@@ -202,7 +202,6 @@ public static boolean toggleIconResizing(boolean toggle) {
      * @return
      */
     public static boolean setAllFiltersOn() {
-        return Microbot.getClientThread().runOnClientThread(() -> {
```
Comment (chsami) [inline runelite-client/src/main/java/net/runelite/client/plugins/microbot/util/settings/Rs2SpellBookSettings.java:205]: Why did you remove this ?

Comment (See1Duck) [reply runelite-client/src/main/java/net/runelite/client/plugins/microbot/util/settings/Rs2SpellBookSettings.java:205]: Because it didn't work, it just failed and didn't execute the code properly whenever I tried to use it
